src/Components/SingleArticle.jsx

```diff
@@ -13,6 +17,49 @@ export const SingleArticle = () => {
 
   const { article_id } = useParams();
 
+  const [hasUpVoted, setHasUpVoted] = useState(false);
```

Comment:
I see what you're doing here, but what if I upvote, downvote, then change my mind and decide I want to upvote again? Why am I not allowed to?

Reply:
I saw that error but I could not think a logic to prevent this error at the moment. I took a note and I'll overcome this problem later.